python_modules/dagster/dagster/_core/definitions/auto_materialize_policy.py

```diff
@@ -53,6 +54,17 @@ class AutoMaterializePolicy(
 
     Constructing an AutoMaterializePolicy directly is not recommended as the API is subject to change.
     AutoMaterializePolicy.eager() and AutoMaterializePolicy.lazy() are the recommended API.
+
+    Args:
+        on_missing (bool): Whether to materialize an asset / partition when it is missing.
```

Comment:
Based on the earlier discussion where we decided to make `eager`/`lazy` the public-facing mental model, I think we should keep some of these args hidden.

Reply:
yeah I'll remove this docstring and then just add this as a parameter to `eager` and `lazy`, with associated docstring on those methods